src/sagemaker/local/image.py

```diff
@@ -98,6 +110,68 @@ def __init__(self, instance_type, instance_count, image, sagemaker_session=None)
         self.container_root = None
         self.container = None
 
+    def process(
+        self, processing_inputs, processing_output_config, environment, processing_job_name
+    ):
+        """Run a processing job locally using docker-compose.
+
+        Args:
+            processing_inputs (dict): The processing input specification.
+            processing_output_config: The processing output configuration specification.
+            environment (dict): The environment collection for the processing job.
+            processing_job_name (str): Name of the local processing job being run.
+        """
+
+        self.container_root = self._create_tmp_folder()
+
+        # A shared directory for all the containers;
+        # it is only mounted if the processing script is Local.
+        shared_dir = os.path.join(self.container_root, "shared")
+        os.mkdir(shared_dir)
+
+        data_dir = self._create_tmp_folder()
+        volumes = self._prepare_processing_volumes(
+            data_dir, processing_inputs, processing_output_config
+        )
+
+        # Create the configuration files for each container that we will create.
+        for host in self.hosts:
+            _create_processing_config_file_directories(self.container_root, host)
+            self.write_processing_config_files(
+                host, environment, processing_inputs, processing_output_config, processing_job_name
+            )
+
+        self._generate_compose_file(
+            "process", additional_volumes=volumes, additional_env_vars=environment
+        )
+        compose_command = self._compose()
+
+        if _ecr_login_if_needed(self.sagemaker_session.boto_session, self.image):
+            _pull_image(self.image)
+
+        process = subprocess.Popen(
+            compose_command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+
+        try:
+            _stream_output(process)
+        except RuntimeError as e:
+            # _stream_output() doesn't have the command line. We will handle the exception
+            # which contains the exit code and append the command line to it.
+            msg = "Failed to run: %s, %s" % (compose_command, str(e))
+            raise RuntimeError(msg)
```

Comment:
maybe better to `raise..from` to preserve original traceback?

```suggestion
        except RuntimeError as e:
            # _stream_output() doesn't have the command line. We will handle the exception
            # which contains the exit code and append the command line to it.
            msg = f"Failed to run: {compose_command}"
            raise RuntimeError(msg) from e
```

Reply:
Done.